packages/mobile/views/dashboard/drawers/activity/ActivityRouter.svelte

```diff
@@ -1,21 +1,49 @@
 <script lang="typescript">
-    import { activityRoute, ActivityRoute } from '../../../../lib/routers'
-    import { StrongholdUnlock } from '../../../../components'
-    import { claimActivity } from '@core/wallet'
+    import { localize } from '@core/i18n'
+    import { isStrongholdUnlocked } from '@core/profile-manager'
+    import { claimActivity, rejectActivity } from '@core/wallet'
+    import { onDestroy } from 'svelte'
+    import { Confirmation, StrongholdUnlock } from '../../../../components'
+    import { activityRoute, ActivityRoute, activityRouter } from '../../../../lib/routers'
     import { selectedActivity } from '../../../../lib/wallet'
+    import { ActivityDetails } from './views'
 
     export let onClose: () => unknown = () => {}
 
-    function onConfirm(): void {
-        $selectedActivity && claimActivity($selectedActivity)
-        onClose()
+    function onReject(): void {
+        if ($activityRoute === ActivityRoute.Reject) {
+            $selectedActivity && rejectActivity($selectedActivity.id)
+            onClose()
+        } else {
+            $activityRouter.next({ route: ActivityRoute.Reject })
+        }
     }
+    async function onClaim(): Promise<void> {
+        const isUnlocked = await isStrongholdUnlocked()
```

Comment:
If possible I think would be good if we can improve this logic here that is also repeated in `packages/mobile/views/dashboard/tabs/ActivityTab.svelte`